src/main/java/htsjdk/samtools/SamReaderFactory.java

```diff
@@ -323,7 +325,25 @@ public SamReader open(final SamInputResource resource) {
                     return reader;
                   }
                 }
-                if (type == InputResource.Type.SEEKABLE_STREAM || type == InputResource.Type.URL) {
+                if (type == InputResource.Type.URI) {
+                    final URI uri = ((UriInputResource) data).uri;
+                    if (uri.getScheme().equalsIgnoreCase("htsget")) {
+                        final URI httpUri = new URI(
+                            "http", uri.getUserInfo(), uri.getHost(), uri.getPort(), uri.getPath(),
```

Comment:
https?

Reply:
This is probably something that needs to be brought up at the next htsget call, but there isn't a way defined in the spec to indicate that a source is an htsget server and we're just using the scheme as a workaround, but this doesn't allow us to specify whether the actual backing source should be http or https. Another workaround to this might be to split the scheme into htsget and htsgets, corresponding to http and https. For the time being this defaults to http because not all servers might support https

Reply:
Will the servers automatically redirect to https supported and you request http?  Defaulting to https seems good because it offers security but possibly more importantly data integrity which I don't think HTTP does?  Maybe we should implement a failover where we try https and then fall back to http if it fails?  Most of the time bam data is private and legally protected so defaulting to http seems a bit scary.

Reply:
Changed to default to https and fall back to http